addons/hr_expense/models/hr_expense.py

```diff
@@ -289,12 +289,14 @@ def _compute_product_uom_id(self):
         for expense in self:
             expense.product_uom_id = expense.product_id.uom_id
 
-    @api.depends('product_id', 'attachment_number')
+    @api.depends('product_id', 'attachment_number', 'currency_rate')
     def _compute_unit_amount(self):
         for expense in self:
             # Only change unit_amount if the product has cost defined on it
             if expense.product_id and expense.product_has_cost and not expense.attachment_number or (expense.attachment_number and not expense.unit_amount):
                 expense.unit_amount = expense.product_id.price_compute('standard_price', currency=expense.currency_id)[expense.product_id.id]
+            else:
+                expense.unit_amount = expense.company_currency_id.round(expense.total_amount_company / (expense.quantity or 1))
```

Comment:
@JulienAlardot Why does not it work without this else?  It seems to go against the comment in L295.

Reply:
The comment was mentioning the old behaviour (removed it and placed one on the else)
The thing is that the unit_amount is used to compute the line balance in the case of a vendor bill (employee paid expense flow)

Reply:
see [L630C1-L641](https://github.com/odoo/odoo/blob/b7a587bd57b73169273bb7c2cab7a23369109fb7/addons/hr_expense/models/hr_expense.py#L630C1-L641)